### PR TITLE
chore: Add loader for email and csat submit buttons

### LINKED ISSUE #2563: bug: Email button loading while submitting the CSAT button
https://user-images.githubusercontent.com/12408980/124438177-7eac0380-dd95-11eb-9df9-fef591a7c518.mov

- Add a separate loader for both buttons. Currently, both buttons are using the same loading prop

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes #2563

### CODE REVIEW

**Review by nithindavid — DISMISSED** [review]

File: app/javascript/shared/components/CustomerSatisfaction.vue [hunk 1]
```diff
@@ -100,11 +104,18 @@ export default {
...
+    async onSubmit() {
+      this.isUpdating = true;
+      await this.$store.dispatch('message/update', {
+        submittedValues: {
+          csat_survey_response: {
+            rating: this.selectedRating,
+            feedback_message: this.feedback,
+          },
+        },
+        messageId: this.messageId,
```
Comment (pranavrajs) [inline app/javascript/shared/components/CustomerSatisfaction.vue:116]: Shouldn't use a try catch here?

Comment (muhsin-k) [reply app/javascript/shared/components/CustomerSatisfaction.vue]: Yes

File: app/javascript/shared/components/CustomerSatisfaction.vue [hunk 2]
```diff
@@ -100,11 +104,23 @@ export default {
...
+          submittedValues: {
+            csat_survey_response: {
+              rating: this.selectedRating,
+              feedback_message: this.feedback,
+            },
+          },
+          messageId: this.messageId,
+        });
+      } catch (error) {
+        // Ignore error
```
Comment (nithindavid) [inline app/javascript/shared/components/CustomerSatisfaction.vue:120]: Shouldn't we be showing error with `this.showAlert`?

Comment (muhsin-k) [reply app/javascript/shared/components/CustomerSatisfaction.vue:120]: @nithindavid I think we are not showing any API related error messages in the widget side

**Review by nithindavid — APPROVED** [review]

**Pull request merged**